src/js/annotations/barplot.js

```diff
@@ -0,0 +1,199 @@
+ * annotations.
+ */
+function getRawBars(chrModels, ideo) {
+    var chr, chrModel, lastBand, numBins, bar, i, px,
+        barWidth = ideo.config.barWidth,
+        bars = [];
+
+    for (chr in chrModels) {
+        chrModel = chrModels[chr];
+        lastBand = chrModel.bands[chrModel.bands.length - 1];
+        numBins = Math.round(lastBand.px.stop / barWidth); // chrPxStop / barWidth
+        console.log(chr, numBins);
+        bar = {chr: chr, annots: []};
+        for (i = 0; i < numBins; i++) {
+            px = i * barWidth - ideo.bump;
+            bar.annots.push({
+                bp: ideo.convertPxToBp(chrModel, px + ideo.bump),
+                px: px,
+                count: 0,
+                chrIndex: chrModel.chrIndex,
+                chrName: chr,
+                color: ideo.config.annotationsColor,
+                annots: []
+            });
+        }
+        bars.push(bar);
+    }
+    return bars;
+}
+
+/**
+ * Assign how many, and which annotations each Barplot bar contains
+ */
+function assignAnnotsToBars(annots, bars, chrModels, ideo) {
+    var chr, chrAnnots, chrModel, barAnnots, i, annot, px, j, barPx, nextBarPx,
+        barWidth = ideo.config.barWidth;
+
+    for (chr in annots) {
+        chrAnnots = annots[chr].annots;
+        chrModel = chrModels[annots[chr].chr]; // get chr by name
+        barAnnots = bars[chrModel.chrIndex].annots;
+        for (i = 0; i < chrAnnots.length; i++) {
+            annot = chrAnnots[i];
+            px = annot.px - ideo.bump;
+            let var_count = annot["count"];
+            for (j = 0; j < barAnnots.length; j++) {
+                barPx = barAnnots[j].px;
+                nextBarPx = barPx + barWidth;
+                if (j === barAnnots.length - 1) nextBarPx += barWidth;
+                if (px >= barPx && px < nextBarPx) {
+                    bars[chrModel.chrIndex].annots[j].count += var_count;
+                    bars[chrModel.chrIndex].annots[j].annots.push(annot);
+                    break;
+                }
+            }
+        }
+    }
+    return bars;
+}
+
+function setIdeoMaxAnnotsPerBar(bars, isFirstGet, ideo) {
+    var maxAnnotsPerBarAllChrs, i, maxAnnotsPerBar, annots, chr, j, barCount;
+
+    if (isFirstGet || ideo.config.BarplotScaling === 'relative') {
+        maxAnnotsPerBarAllChrs = 0;
+        for (i = 0; i < bars.length; i++) {
+            maxAnnotsPerBar = 0;
+            annots = bars[i].annots;
+            chr = bars[i].chr;
+            for (j = 0; j < annots.length; j++) {
+                barCount = annots[j].count;
+                if (barCount > maxAnnotsPerBar) maxAnnotsPerBar = barCount;
+                if (barCount > maxAnnotsPerBarAllChrs) {
+                    maxAnnotsPerBarAllChrs = barCount;
+                }
+            }
+            ideo.maxAnnotsPerBar[chr] = maxAnnotsPerBar;
+        }
+        ideo.maxAnnotsPerBarAllChrs = maxAnnotsPerBarAllChrs;
+    }
+}
```

Comment:
This function is identical to [`setIdeoMaxAnnotsPerBar` in src/js/annotations/histogram.js](https://github.com/eweitz/ideogram/blob/8ecea503da29e98ef6434acfae2981c68023f9c4/src/js/annotations/histogram.js#L61).  Like other cases here, please import it instead of copying and pasting it here.